llvm/lib/Target/RISCV/RISCVMacroFusion.td

```diff
@@ -0,0 +1,21 @@
+//==----- RISCVMacroFusion.td - Macro Fusion Definitions -----*- tablegen -*-=//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// ===---------------------------------------------------------------------===//
+// The following definitions describe the macro fusion predicators.
+
```

Comment:
If we had a file `MacroFusionUtils` which contained same helper functions like `checkIsVRegOperaned`, would the same code written in C++ be much longer?

Reply:
The `MacroFusionUtils` can be useful to simplify TableGen-generated code. It's a good idea, I will try to implement this later!